### ISSUE TITLE
Server stalls when multiple ppl download big mods

### ISSUE THREAD

**Issue by Cicrocoft** [StrangeLoopGames/EcoIssues#20600]
There seems to be an issue with adding large sized mods to the servers. When multiple (2 and up) players joins the server and needs to download them, latency goes through the roof (+8000ms) for the logged in players, and the ppl joining gets stuck at "downloading mods". It seems like mods around 30mb is max, as these downloads ok. 

Issue is related to Elixirs large mods (might be the only mods of that size?)

I've attached a screenshot of what the server host thinks is the issue, and another screenshot from the DL dev and his experience with it. ![image](https://user-images.githubusercontent.com/20983572/108106403-2f875f80-708e-11eb-9c3c-441c6765b364.jpeg)![image](https://user-images.githubusercontent.com/20983572/108106415-331ae680-708e-11eb-92bd-7f41b426a1df.jpeg)

**Comment by MonzUn** [comment]
I'm the author of the referenced Discord message.

Just wanted to say that there are more people seeing this issue than OP.

I have about 200mbps and I tested the network throughput while this issue was active and it was fine.

**Comment by MishkaFur** [comment]
My server is also experiencing this (24mb mod) whilst a new user is connecting the whole server hangs until the download is done so does not seem to be related to huge mods but in general any

Also gigabit connection and no issues connection wise

**Comment by E-4forlife** [comment]
+1 on this for my server as well when more then one person is joining for the first time and trying to download mods the whole server client locks up the hardware the server is on dose not go over 30-40% usage of everything and it has a gig line and dose not go over 10-15 mbs

**Comment by drkshrk** [comment]
We have gotten this on server as well - more prevalent when multiple people are connecting at the same time.

**Comment by mirasrael** [comment]
For 9.4.1 available test build: https://drive.google.com/file/d/1o5ifgAEIjhAUabI0hI7fZevmXFcIybsU/view?usp=sharing

**Comment by SlayksWood** [comment]
Tested with Khairi, seems fixed.

**Comment by Dave247** [comment]
This is not fixed and is still happening on 9.6.1, now being the second time we have experienced this issue, with the previous being on 9.4.3 I believe, both with different hosts. 

Based on what little info we have been able to gather, the assumption is that the download process while being rather crap is not fit for purpose. Two possibilities/assumptions we have made is that the downloads are synchronous and not asynchronous, or that the issue will tigger if the geo-location of the client differs too greatly from the server.
